connectors/sources/sharepoint.py

```diff
@@ -238,14 +241,105 @@ async def get_content(
+        """
+        response = await anext(
+            self._api_call(
+                url_name=LIST_BY_TITLE,
+                host_url=self.host_url,
+                parent_site_url=site_url,
+                filename=filename,
+            )
+        )
+        return response["value"][0].get("CanvasContent1")  # pyright: ignore
+
+    async def convert_file_to_b64(self, source_file_name):
+        """This method convert th file content into b64
+        Args:
+            source_file_name: Name of source file
+        Returns:
+            attachment_content: Attachment content in b64
+        """
+        async with aiofiles.open(file=source_file_name) as target_file:
             # base64 on macOS will add a EOL, so we strip() here
             attachment_content = (await target_file.read()).strip()
-        await remove(source_file_name)  # pyright: ignore
+        try:
+            await remove(source_file_name)  # pyright: ignore
+        except Exception as exception:
+            logger.warning(
+                f"Could not remove file from: {source_file_name}. Error: {exception}"
+            )
+        return attachment_content
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def get_site_pages_content(
+        self, document, site_url, list_response, timestamp=None, doit=False
+    ):
+        """Get content of site pages for SharePoint
+
+        Args:
+            document (dictionary): Modified document.
+            site_url (str): Site path of sharepoint
+            list_response (dict): Dictionary of list item response
+            timestamp (timestamp, optional): Timestamp of item last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            dictionary: Content document with id, timestamp & text.
+        """
+        document_size = int(document["size"])
+        filename = (
+            document["title"] if document["type"] == "File" else document["file_name"]
+        )
+        if not (doit and document_size):
+            return
+
+        if document_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {document_size} of file {filename} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+
+        source_file_name = ""
+        if self.is_cloud:
+            response_data = await self.get_site_page_for_online(site_url, filename)
+        else:
+            response_data = list_response["WikiField"]
+
+        if response_data is None:
+            return
+
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            await async_buffer.write(bytes(response_data, "utf-8"))
+
+            source_file_name = async_buffer.name
+
+        await asyncio.to_thread(
+            convert_to_b64,
```

Comment:
Why do you convert the result into base64? `response_data ` is a raw HTML of the rendered Site Page right?

Reply:
when I try to pass the content directly to attachment field , It showing error on elastic saying: [ "Processor 'attachment' in pipeline 'ent-search-generic-ingestion' failed with message 'Illegal base64 character 3c'" ] and when I try to pass it in body directly then complete html content is indexing on elastic with the tags(attached the screenshot).  Initially with base64 conversion we were getting the data without tags attached.
![Screenshot (87)](https://user-images.githubusercontent.com/125074214/233355678-74c3d1ae-0f09-42ae-a560-da1a82724166.png)
![Screenshot (88)](https://user-images.githubusercontent.com/125074214/233355694-d9407af1-9324-4349-a83b-e2be3e99880e.png)

Reply:
Do you mean that passing `base64` encoded raw content into `_attachemnt` will result to a content without HTML tags in `body`?